src/portal_to_terra_input_from_anaset.py

```diff
@@ -0,0 +1,298 @@
+    '/platform-terms/EFO_0008637/_RNA-seq_rna': {'R1': 'read1', 'R2': 'read2'},     # For test analysis set only
+    '/platform-terms/EFO_0010963/_ATAC-seq_atac': {'R1': 'read1', 'R2': 'read2'},    # For test analysis set only
+    '/platform-terms/NTR_0000765/_10x multiome with MULTI-seq_rna': {'R1': 'read1', 'R2': 'read2'},
+    '/platform-terms/NTR_0000765/_10x multiome with MULTI-seq_atac': {'R1': 'read1', 'R3': 'read2'},
+    '/platform-terms/EFO_0010963/_Parse SPLiT-seq_rna': {'R1': 'read1', 'R2': 'read2'}
+}
+
+
+def get_assay_type_from_assay_id(assay_term_id: str, igvf_api) -> str:
+    # NOTE: This may get more complicated if the pipeline wants to distinguish sn vs. sc
+    """Infer assay type (RNA or ATAC) from the assay term summary
+
+    Args:
+        assay_term_id (str): assay term id
+        igvf_api (_type_): _description_
+
+    Returns:
+        str: rna or atac
+    """
+    assay_term_summary = igvf_api.get_by_id(assay_term_id).actual_instance.summary
+    if 'RNA' in assay_term_summary:
+        return 'rna'
+    elif 'ATAC' in assay_term_summary:
+        return 'atac'
+
+
+def get_samp_taxa_from_samp_id(sample_id: str, igvf_api) -> str:
+    """Get sample taxa from a sample ID
+
+    Args:
+        sample_id (str): e.g., /in-vitro-system/IGVFxxxxx
+        igvf_api (_type_): _description_
+
+    Returns:
+        str: e.g., Homo sapiens
+    """
+    sample_item = igvf_api.get_by_id(sample_id).actual_instance
+    return sample_item.taxa
+
+
+def get_measet_ids_and_other_info_from_analysis_set_acc(analysis_set_accessions: list, igvf_api) -> tuple:
+    """Get all input measurement set IDs and sample taxa from an analysis set.
+
+    Args:
+        analysis_set_accessions (list): A list of one analysis set accession
+        igvf_api (_type_): _description_
+
+    Returns:
+        tuple: (list(taxa), list(measurement ids))
+    """
+    analysis_set_obj = igvf_api.analysis_sets(accession=analysis_set_accessions).graph[0]
+    measurement_ids = [measet_id for measet_id in analysis_set_obj.input_file_sets if 'measurement-sets' in measet_id]
+    # lab_id = analysis_set_obj.lab
+    # award_id = analysis_set_obj.award
+    # TODO: Are we expecting different taxa in a single Analysis set?
+    sample_taxa = list(set([get_samp_taxa_from_samp_id(sample_id=samp_id, igvf_api=igvf_api) for samp_id in analysis_set_obj.samples]))
+    return (sample_taxa, measurement_ids)
+
+
+def get_and_sort_measurement_sets_by_assays(measurement_set_ids: list, igvf_api) -> list:
+    """Sort a list of measurement set items based on assay terms.
+
+    Args:
+        measurement_set_ids (list): A list of measurement set items generated by the APi.
+        igvf_api (_type_): _description_
+
+    Returns:
+        list: Same content as the input list but sorted by the assay terms.
+    """
+    return sorted([igvf_api.get_by_id(measet_id).actual_instance for measet_id in measurement_set_ids], key=lambda x: x.assay_term)
```

Comment:
need to be a little careful here, this won't scale for more than a few measurement_sets.    You can use a single search to get an arbitrary list of objects, something like &accession=XXX122&accession=YYY233&accession=...  but this also has limites (~8000 character URIs)